File: 120-triangle/120-triangle.py
class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        
        tLen = len(triangle)
        
        if (tLen == 1): return triangle[0][0];
        
        for i, row in enumerate(triangle):
            
            if i == 0: continue;
            
            for j, val in enumerate(row):
                
                if j == 0: row[j] += triangle[i - 1][j]
                elif i == j: row[j] += triangle[i - 1][j - 1]
                else: 
                    row[j] += min(triangle[i - 1][j - 1], triangle[i - 1][j])
                    
        return min(triangle[tLen - 1])
            
                    
    
    

File: 120-triangle/120-triangle.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        
        if (triangle.size() == 1) return triangle[0][0];
        
        int len = triangle.size();
        
        for (int i = 1; i < len; i++) {
            
            for (int j = 0; j <= i; j++) {
                
                if (j == 0) {
                    triangle[i][j] += triangle[i - 1][j];
                }
                else if (j == i) {
                    triangle[i][j] += triangle[i - 1][j - 1];
                }
                else {
                    triangle[i][j] += min(triangle[i - 1][j], triangle[i - 1][j - 1]);
                }
            }
        }
        
        int ret = 111111;
        
        for (int i = 0; i < len; i++) {
            ret = min(ret, triangle[len - 1][i]);
        }
        
        return ret;
    }
};